unity-environment/Assets/ML-Agents/Scripts/Batcher.cs

```diff
@@ -0,0 +1,279 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Google.Protobuf;
+
+namespace MLAgents
+{
+
+    public class Batcher
+    {
+        const int NUM_AGENTS = 32;
+        Dictionary<string, bool> hasSentState =
```

Comment:
replace string with `BrainId`

Reply:
Just to make sure I understand you @flamz3d, you are suggesting inferring the type of the variable from the declaration rather than implicit typing (which I don't think is available in C# anyways)? 

This applies more to the Python side of things where I think some type hints would improve readability significantly.

Reply:
I mean use C# type inference:
```
var hasSentState = new Dictionary<string, bool>();
```
instead of
```
Dictionary<string, bool> hasSentState = new Dictionary<string, bool>();
```